py/desispec/correct_cte.py

```diff
@@ -542,71 +569,86 @@ def get_cte_images(night, camera, expids=None):
     -------
     List of preprocessed Images without CTE correction applied
     """
+    # inside func to avoid circular import
+    import desispec.io
+    import desispec.io.util
 
     log = get_logger()
 
-    if expids is None:
+    if expids is not None:
+        filename = desispec.io.findfile('raw', night, expids[0])
+        header   = fitsio.read_header(filename, camera.upper())
+        if not needs_ctecorr(header):
+            log.info(f"No CTE correction needed for {night} {camera}")
+            return None
+
+    else:
+        #- Look up exposures in production exposure table
         exptablefn = os.path.join(os.environ['DESI_SPECTRO_REDUX'],
                                   os.environ['SPECPROD'],
                                   'exposure_tables', str(night // 100),
                                   f'exposure_table_{night}.csv')
 
         if not os.path.isfile(exptablefn) :
             mess = f"Cannot find exposure table file '{exptablefn}'. Because of that the flat exposures needed for the CTE correction modeling cannot be identified. Maybe check env. variables DESI_SPECTRO_REDUX and SPECPROD?"
-            log.error(mess)
+            log.critical(mess)
             raise RuntimeError(mess)
 
-        #- Read exptable and apply quality cuts to flats
+        #- Read exptable and keep just the good flats
         exptable = Table.read(exptablefn)
         keep = exptable['OBSTYPE'] == 'flat'
         keep &= exptable['LASTSTEP'] != 'ignore'
         exptable = exptable[keep]
 
+        #- First check 1 sec flat header to see if we even need to do CTE corr
+        selection = (np.abs(exptable['EXPTIME'] - 1) < 0.1)
+        if np.sum(selection)<1 :
+            mess = f"No 1 sec flat for {night} (in {exptablefn}), needed for CTE correction model fit"
+            log.critical(mess)
+            raise RuntimeError(mess)
+
+        filename = desispec.io.findfile('raw', night, exptable['EXPID'][selection][0])
+        header   = fitsio.read_header(filename, camera.upper())
+        if not needs_ctecorr(header):
+            log.info(f"No CTE correction needed for {night} {camera}")
+            return None
+
+        #- CTE correction is needed, so also filter by CAMWORD and BADCAMWORD
         keep = np.ones(len(exptable), dtype=bool)
         for i in range(len(keep)):
-            if camera not in decode_camword(str(exptable['CAMWORD'][i])):
+            if camera not in desispec.io.util.decode_camword(str(exptable['CAMWORD'][i])):
                 keep[i] = False
-            elif camera in decode_camword(str(exptable['BADCAMWORD'][i])):
+            elif camera in desispec.io.util.decode_camword(str(exptable['BADCAMWORD'][i])):
```

Comment:
In the future, after `pipelinerefactor` is merged, `desispec.io.util.all_impacted_cameras(badcamword, badamps)` returns the set of cameras reflected in either badcamword or badamps, where each is a single string. You could change this to:
`if camera in desispec.io.util.all_impacted_cameras(str(exptable['BADCAMWORD'][i]), str(exptable['BADAMPS'][i])):`

But that obviously won't work for the interim. For now you could leverage the fact that BADAMPS is a comma-seperated list with cameras in the identifiers, so e.g `'b4' in 'z1A,b4B,z7C'` is `True`, as desired.

So you can do the following:
`if camera in desispec.io.util.decode_camword(str(exptable['BADCAMWORD'][i])) or camera in str(exptable['BADAMPS'][i]):`